### PR TITLE
Track rss and sitemap

### PR DESCRIPTION
**Pull request by finestructure**

### CODE REVIEW

File: Sources/App/Core/AppEnvironment.swift [hunk 1]
```diff
@@ -179,7 +189,7 @@ extension AppEnvironment {
         },
         mastodonPost: Mastodon.post(client:message:),
         metricsPushGatewayUrl: { Environment.get("METRICS_PUSHGATEWAY_URL") },
-        plausibleAPIReportingSiteID: { Environment.get("PLAUSIBLE_API_REPORTING_SITE_ID") },
+        plausibleBackendReportingSiteID: { Environment.get("PLAUSIBLE_BACKEND_REPORTING_SITE_ID") },
```
Comment (finestructure) [inline Sources/App/Core/AppEnvironment.swift:192]: I've added the corresponding new variables to `dev` and `prod` and will remove the old one after deployment.

File: Sources/App/Core/Plausible.swift [hunk 2]
```diff
@@ -31,16 +31,9 @@ enum Plausible {
...
-    }
-
-    enum Identifier {
-        case anonymous
-        case identifier(String)
-
-        var props: [String: String] {
-            fatalError("fix me")
-        }
+        case sitemap = "/sitemap"
```
Comment (finestructure) [inline Sources/App/Core/Plausible.swift:36]: These (`/rss` and `/sitemap`) aren't real paths, they're just for filtering in Plausible. I've just picked something that seemed sensible.

File: Sources/App/routes.swift [hunk 3]
```diff
@@ -259,16 +259,20 @@ func routes(_ app: Application) throws {
...
+            
+            $0.get(SiteURL.rssReleases.pathComponents, use: RSSFeed.showReleases)
+                .excludeFromOpenAPI()
+        }
 
-        app.get(SiteURL.siteMap.pathComponents) { req in
-            SiteMap.fetchPackages(req.db)
-                .map(SiteURL.siteMap)
-        }.excludeFromOpenAPI()
+        app.group(BackendReportingMiddleware(path: .sitemap)) {
```
Comment (daveverwer) [inline Sources/App/routes.swift:270]: When I rebase #2416 on top of this I'll create separate tracking for the three different site map types as we want to know how often it hits the index vs the package-specific site maps.

**Review by daveverwer — APPROVED** [review]

File: Sources/App/Core/Plausible.swift (hunk 2 above)
Comment (daveverwer) [reply Sources/App/Core/Plausible.swift:36]: Yes this is fine. No need for the `.xml`.

File: Sources/App/routes.swift [hunk 4]
```diff
@@ -259,16 +259,20 @@ func routes(_ app: Application) throws {
...
-
-        app.get(SiteURL.rssReleases.pathComponents, use: RSSFeed.showReleases)
-            .excludeFromOpenAPI()
+        app.group(BackendReportingMiddleware(path: .rss)) {
+            $0.get(SiteURL.rssPackages.pathComponents, use: RSSFeed.showPackages)
+                .excludeFromOpenAPI()
+            
+            $0.get(SiteURL.rssReleases.pathComponents, use: RSSFeed.showReleases)
+                .excludeFromOpenAPI()
+        }
```
Comment (finestructure) [inline Sources/App/routes.swift:268]: I've collapsed both rss feed paths (the actual ones are `/packages.rss` and `releases.rss` into `/rss` for sake of simplicity. We could also report them individually under some path, perhaps with a `/rss/` prefix to make filtering easier 🤷‍♂️

Comment (daveverwer) [reply Sources/App/routes.swift:268]: I think this is fine as one trigger. No benefit in splitting these out. 👍

**Pull request merged**